ENGAGEHF/Home.swift

```diff
@@ -18,27 +20,43 @@ struct HomeView: View {
     static var accountEnabled: Bool {
         !FeatureFlags.disableFirebase && !FeatureFlags.skipOnboarding
     }
+    
+    private var bluetoothEnabled: Bool {
+        !ProcessInfo.processInfo.isPreviewSimulator
+    }
 
-
+    @Environment(MeasurementManager.self) private var measurementManager
+    @Environment(WeightScaleDevice.self) private var weightScale: WeightScaleDevice?
+    @Environment(Bluetooth.self) private var bluetooth
+    
     @AppStorage(StorageKeys.homeTabSelection) private var selectedTab = Tabs.home
     @State private var presentingAccount = false
-
+    
     
     var body: some View {
+        @Bindable var measurementManager = measurementManager
+        
         TabView(selection: $selectedTab) {
             Dashboard(presentingAccount: $presentingAccount)
                 .tag(Tabs.home)
                 .tabItem {
                     Label("Home", systemImage: "house")
                 }
         }
+            .autoConnect(enabled: weightScale == nil && bluetoothEnabled, with: bluetooth)
             .sheet(isPresented: $presentingAccount) {
                 AccountSheet()
             }
             .accountRequired(Self.accountEnabled) {
                 AccountSheet()
             }
             .verifyRequiredAccountDetails(Self.accountEnabled)
+            .sheet(isPresented: $measurementManager.showSheet, onDismiss: {
+                MeasurementManager.manager.clear()
+            }) {
+                MeasurementRecordedView()
+                    .presentationDetents([.fraction(0.45)])
+            }
```

Comment:
I would generally prefer a formatting like this, similar to the feedback for the buttons:
```suggestion
            .sheet(
                isPresented: $measurementManager.showSheet,
                onDismiss: {
                    MeasurementManager.manager.clear()
                },
                content: {
                    MeasurementRecordedView()
                        .presentationDetents([.fraction(0.45)])
                }
            )
```